picard/formats/__init__.py

```diff
@@ -49,6 +71,8 @@ def open(filename):
         format = _extensions[ext]
     except KeyError:
         return None
```

Comment:
@samj1912 : according to this code, if extension isn't found, KeyError is raised and `open()` returns None
So there's no attempt of using the file magic number detection.
If another exception is raised, then we use `guess_format()`, but if it fails we raise an exception ("Unknown audio format").
I may miss something, but i don't hink it is the behavior we expect, and return/exception should be consistent whichever method we use.

Since using extension is prolly faster, we should fallback on magic number method if the extension isn't known, but if the format cannot be identified using this method, still return None.